src/net/connect.cpp

```diff
@@ -157,8 +156,44 @@ void connect_ssl(bufferevent *orig_bev, ssl_st *ssl,
     }
 
     bufferevent_setcb(bev, nullptr, nullptr, mk_bufferevent_on_event,
-            new Callback<bufferevent *>([cb, logger](Error err, bufferevent *bev) {
+            new Callback<bufferevent *>([cb, logger, ssl, hostname](Error err, bufferevent *bev) {
```

Comment:
I don't think `ssl` should be passed in the lambda capture list, as we get it from the bufferevent below.